python/cudf/cudf/_lib/cpp/exception_handler.hpp

```diff
@@ -0,0 +1,80 @@
+/*
```

Comment:
The `cudf/_lib/cpp` directory contains the Cython definitions ("declarations") for `libcudf` types and functions.

Perhaps we can place this (and the corresponding `.pxd` file) in `cudf/_lib/exception_handler.hpp` (and `cudf/_lib/exception_handler.pxd`) instead?

Alternately, I'd also be fine with just inlining this bit of C++ in `cudf/_lib_exception_handler.pxd`

Reply:
Yeah, I wasn't sure what the place was for this given the C++. Personally I prefer avoiding inline C++ in pxd files, but I can go that route if you like it. IMO it will be familiar to an even smaller subset of devs, and no editors are capable of syntactically or semantically parsing it as anything other than a comment. Moving it to `_lib` is also fine with me; it seemed somewhat appropriate to put in `cpp` since `exception_handler.pxd` is now also just a file of Cython "declarations" rather than new definitions, but on further consideration I think `_lib` is more appropriate too.

Reply:
OK let's move them both to `_lib` (and keep them separate)

Reply:
I've made this change, but one thing to note is that it does make for some slightly awkward circularity. We are now importing the handler from `_lib` into all the pxd files in `_lib/cpp` (because the latter is where we actually expose C++ APIs that need to be decorated with +cudf_exception_handler). Do you still like that outcome?

Reply:
I think that's fine